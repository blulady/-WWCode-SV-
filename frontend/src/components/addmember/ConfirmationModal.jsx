import React from "react";

const ConfirmationModal = (props) => {
  return (
    <div className="modal" id="confirmModal" tabIndex="-1" role="dialog">
      <div className="modal-dialog modal-dialog-centered">
        <div className="modal-content">
          <div className="modal-body modal-open-text">
            *Selected role for this member is{" "}
            <span className="confirm-message">{props.memberrole}</span>, who
            will have{" "}
            <span className="confirm-message">{props.memberdesc}</span>
            <div className="d-flex justify-content-center">
              <button
                type="button"
                className="modal-confirm-btn"
                data-bs-dismiss="modal"
              >
                Confirm
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
export default ConfirmationModal;
