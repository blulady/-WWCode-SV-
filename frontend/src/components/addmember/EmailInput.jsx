import React from "react";
import EditIcon from "../../images/edit_24px.png";

const EmailInput = (props) => {
  return (
    <div className="div-inline responsive-div-block">
      <p className={props.pclass}>{props.emailvalue}</p>
      <input
        type="email"
        className={props.className}
        id="email"
        name={props.name}
        value={props.value}
        onChange={props.onChange}
        placeholder={props.placeholder}
        required
      />
      <div className="spacer" />
      <div className="edit-icon">
        <img
          className={props.editclass}
          src={EditIcon}
          style={{ width: "auto", height: "auto" }}
          alt="Edit"
          onClick={props.editiconclick}
        />
      </div>
      {/* <button  className={props.buttonclass} onClick={props.savebuttonclick}> */}
      <button type="submit" className={props.buttonclass} onClick={props.savebuttonclick}>
        Save
      </button>
    </div>
  );
};
export default EmailInput;
