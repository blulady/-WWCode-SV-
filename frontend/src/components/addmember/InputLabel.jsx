import React from "react";

const InputLabel = (props) => {
  return (
    <div>
      <label htmlFor={props.labelname} className="input-label">
        {props.text}
      </label>
    </div>
  );
};
export default InputLabel;
