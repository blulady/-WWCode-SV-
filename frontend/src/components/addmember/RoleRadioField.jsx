import React from "react";

const RoleRadioInput = (props) => {
  return (
    <>
      <input
        checked={props.checked}
        type="radio"
        className={props.className}
        id={props.id}
        name={props.name}
        value={props.value}
        data-bs-toggle="modal"
        data-bs-target={props.datatarget}
        onClick={props.onClick}
        onChange={props.onChange}
      />
      <div className="member-role">{props.roletext}</div>
      <div className="role-desc">{props.roledesc}</div>
    </>
  );
};
export default RoleRadioInput;
