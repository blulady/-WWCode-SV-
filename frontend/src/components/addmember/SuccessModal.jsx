import React from "react";

const SuccessModal = (props) => {
  return (
    <div
      className="modal modal-success-bg show"
      id="successModal"
      tabIndex="-1"
      role="dialog"
    >
      <div className="modal-dialog modal-dialog-centered">
        <div className="modal-content">
          <div className="modal-header modal-success-header">
            <button
              type="button"
              className="close"
              aria-label="Close"
              data-dismiss="modal"
              close={props.close}
            >
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
          <div className="modal-body modal-success-body">
            <div className="modal-success-title">Success!</div>
            <div className="modal-success-body-text">
              {" "}
              Member has been added and emailed a registration link
            </div>
            <div className="d-flex justify-content-center">
              <button
                type="button"
                className="modal-success-btn"
                data-dismiss="modal"
                onClick={props.onClick}
              >
                Close
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
export default SuccessModal;
