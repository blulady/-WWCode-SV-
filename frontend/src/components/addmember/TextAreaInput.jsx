import React from "react";
import EditIcon from "../../images/edit_24px.png";

const TextAreaInput = (props) => {
  return (
    <div className="div-inline responsive-div-block">
      <p className={props.memberclass}>{props.message}</p>
      <div style={{ display: "block" }}>
        <textarea
          className={props.className}
          id="message"
          name={props.name}
          value={props.value}
          rows="6"
          placeholder="Add an optional message to the registration email sent to the user (max 2000 char)"
          onChange={props.onChange}
          maxLength="2000"
          data-testid='message'
        ></textarea>
        <div className={props.counterclass}>{props.countervalue}</div>
      </div>
      <div className="spacer" />
      <div className="edit-icon">
        <img
          className={props.editclass}
          src={EditIcon}
          style={{ width: "auto", height: "auto" }}
          alt="Edit"
          onClick={props.editiconclick}
        />
      </div>
      <button type= "submit" className={props.buttonclass} onClick={props.savebuttonclick}>
        Save
      </button>
    </div>
  );
};
export default TextAreaInput;
