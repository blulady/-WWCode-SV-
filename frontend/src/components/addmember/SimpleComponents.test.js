import React from 'react';
import { render, fireEvent, act } from '@testing-library/react';
import ConfirmationModal from './ConfirmationModal';
import EmailInput from './EmailInput';
import InputLabel from './InputLabel';
import RoleRadioField from './RoleRadioField';
import SuccessModal from './SuccessModal';
import TextAreaInput from './TextAreaInput';

describe('ConfirmationModal', () => {
    const mockShowModal = jest.fn();

    test('it renders without crashing', () => {
        const { container } = render(<ConfirmationModal memberrole='Volunteer'
                                                        memberdesc='Limited access to areas of portal'
                                                        onClick={mockShowModal} />);
        
        expect(container).toMatchSnapshot();
    });

    test('it calls onClick handler on confirmation click', () => {
        const { container } = render(<ConfirmationModal memberrole='Volunteer'
                                                        memberdesc='Limited access to areas of portal'
                                                        onClick={mockShowModal} />);
        const button = container.querySelector('.modal-confirm-btn');
        
        fireEvent.click(button);

        expect(mockShowModal).toHaveBeenCalledTimes(1);
    });
});

describe('EmailInput', () => {
    const mockHandleChange = jest.fn();

    test('it renders without crashing', () => {
        const { container } = render(<EmailInput name='Email'
                                                 className='form-control email-input'
                                                 pclass='hide'
                                                 editclass='hide'
                                                 buttonclass='hide'
                                                 onChange={mockHandleChange}
                                                 placeholder='eg. sam@wwcode.com' />);
        
        expect(container).toMatchSnapshot();
    });

    test('it calls onChange handler on input change', () => {
        const { container } = render(<EmailInput name='Email'
                                                 className='form-control email-input'
                                                 pclass='hide'
                                                 editclass='hide'
                                                 buttonclass='hide'
                                                 onChange={mockHandleChange}
                                                 placeholder='eg. sam@wwcode.com' />);
        const input = container.querySelector('.email-input');

        fireEvent.change(input, { target: { value: 'test@example.com' } });
        
        expect(mockHandleChange).toHaveBeenCalledTimes(1);
        expect(input.value).toBe('test@example.com');
    });
});

describe('InputLabel', () => {
    test('it renders without crashing', () => {
        const { container } = render(<InputLabel labelname='test' text='hello world!' />);
        
        expect(container).toMatchSnapshot();
    });
});

describe('RoleRadioField', () => {
    const mockHandleClick = jest.fn();
    const mockHandleChange = jest.fn();

    test('it renders without crashing', () => {
        const { container } = render(<RoleRadioField className='form-check-input position-static role-radio'
                                                     id='1'
                                                     name='Role'
                                                     value='Volunteer'
                                                     pclass='hide'
                                                     datatarget='#confirmModal'
                                                     onClick={mockHandleClick}
                                                     onChange={mockHandleChange}
                                                     roletext='Volunteer'
                                                     roledesc='Limited access to areas of portal' />);
        
        expect(container).toMatchSnapshot();
    });

    test('it calls onClick handler on input click', () => {
        const { container } = render(<RoleRadioField className='form-check-input position-static role-radio'
                                                     id='1'
                                                     name='Role'
                                                     value='Volunteer'
                                                     pclass='hide'
                                                     datatarget='#confirmModal'
                                                     onClick={mockHandleClick}
                                                     onChange={mockHandleChange}
                                                     roletext='Volunteer'
                                                     roledesc='Limited access to areas of portal' />);
        const input = container.querySelector('.role-radio');
                                                
        fireEvent.click(input);

        expect(mockHandleClick).toHaveBeenCalledTimes(1);
    });
});

describe('SuccessModal', () => {
    const mockHandleClick = jest.fn();

    test('it renders without crashing', () => {
        const { container } = render(<SuccessModal onClick={mockHandleClick} />);
        
        expect(container).toMatchSnapshot();
    });

    test('it calls onClick handler on button click', () => {
        const { container } = render(<SuccessModal onClick={mockHandleClick} />);
        const button = container.querySelector('.modal-success-btn');

        fireEvent.click(button);

        expect(mockHandleClick).toHaveBeenCalledTimes(1);
    });
});

describe('TextAreaInput', () => {
    let mockMessage = "Hello World!";
    let mockNewMember = {
        Email: '',
        Role: '',
        Message: mockMessage
    }
    const mockHandleChange = jest.fn().mockImplementation((e) => {
        mockNewMember.Message = e.target.value;
    });
    
    test('it renders without crashing', () => {
        const { container } = render(<TextAreaInput name='Message'
                                                    pclass='hide'
                                                    editclass='hide'
                                                    buttonclass='hide'
                                                    className='form-control message-textarea'
                                                    onChange={mockHandleChange}
                                                    counterclass='message-counter'
                                                    value={mockNewMember.message}
                                                    countervalue={ mockNewMember.Message
                                                        ? mockNewMember.Message.length + "/2000 char"
                                                        : null} />);
        
        expect(container).toMatchSnapshot();
    });

    test('it calls onChange handler on textarea change', () => {
        mockMessage = 'Hello There World!'
        const { container } = render(<TextAreaInput name='Message'
                                                    pclass='hide'
                                                    editclass='hide'
                                                    buttonclass='hide'
                                                    className='form-control message-textarea'
                                                    onChange={mockHandleChange}
                                                    counterclass='message-counter'
                                                    value={mockNewMember.message}
                                                    countervalue={ mockNewMember.Message
                                                        ? mockNewMember.Message.length + "/2000 char"
                                                        : null} />);
        const textArea = container.querySelector('.message-textarea');
        
        fireEvent.change(textArea, { target: { value: mockMessage } });

        expect(mockHandleChange).toHaveBeenCalledTimes(1);
        expect(textArea.value).toBe('Hello There World!');
    });
});