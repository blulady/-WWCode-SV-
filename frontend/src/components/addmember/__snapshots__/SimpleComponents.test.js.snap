// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal it renders without crashing 1`] = `
<div>
  <div
    class="modal"
    id="confirmModal"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-dialog-centered"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-body modal-open-text"
        >
          *Selected role for this member is
           
          <span
            class="confirm-message"
          >
            Volunteer
          </span>
          , who will have
           
          <span
            class="confirm-message"
          >
            Limited access to areas of portal
          </span>
          <div
            class="d-flex justify-content-center"
          >
            <button
              class="modal-confirm-btn"
              data-bs-dismiss="modal"
              type="button"
            >
              Confirm
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EmailInput it renders without crashing 1`] = `
<div>
  <div
    class="div-inline responsive-div-block"
  >
    <p
      class="hide"
    />
    <input
      class="form-control email-input"
      id="email"
      name="Email"
      placeholder="eg. sam@wwcode.com"
      required=""
      type="email"
      value=""
    />
    <div
      class="spacer"
    />
    <div
      class="edit-icon"
    >
      <img
        alt="Edit"
        class="hide"
        src="edit_24px.png"
        style="width: auto; height: auto;"
      />
    </div>
    <button
      class="hide"
      type="submit"
    >
      Save
    </button>
  </div>
</div>
`;

exports[`InputLabel it renders without crashing 1`] = `
<div>
  <div>
    <label
      class="input-label"
      for="test"
    >
      hello world!
    </label>
  </div>
</div>
`;

exports[`RoleRadioField it renders without crashing 1`] = `
<div>
  <input
    class="form-check-input position-static role-radio"
    data-bs-target="#confirmModal"
    data-bs-toggle="modal"
    id="1"
    name="Role"
    type="radio"
    value="Volunteer"
  />
  <div
    class="member-role"
  >
    Volunteer
  </div>
  <div
    class="role-desc"
  >
    Limited access to areas of portal
  </div>
</div>
`;

exports[`SuccessModal it renders without crashing 1`] = `
<div>
  <div
    class="modal modal-success-bg show"
    id="successModal"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-dialog-centered"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header modal-success-header"
        >
          <button
            aria-label="Close"
            class="close"
            data-bs-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
          </button>
        </div>
        <div
          class="modal-body modal-success-body"
        >
          <div
            class="modal-success-title"
          >
            Success!
          </div>
          <div
            class="modal-success-body-text"
          >
             
            Member has been added and emailed a registration link
          </div>
          <div
            class="d-flex justify-content-center"
          >
            <button
              class="modal-success-btn"
              data-dismiss="modal"
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TextAreaInput it renders without crashing 1`] = `
<div>
  <div
    class="div-inline responsive-div-block"
  >
    <p />
    <div
      style="display: block;"
    >
      <textarea
        class="form-control message-textarea"
        data-testid="message"
        id="message"
        maxlength="2000"
        name="Message"
        placeholder="Add an optional message to the registration email sent to the user (max 2000 char)"
        rows="6"
      />
      <div
        class="message-counter"
      >
        12/2000 char
      </div>
    </div>
    <div
      class="spacer"
    />
    <div
      class="edit-icon"
    >
      <img
        alt="Edit"
        class="hide"
        src="edit_24px.png"
        style="width: auto; height: auto;"
      />
    </div>
    <button
      class="hide"
      type="submit"
    >
      Save
    </button>
  </div>
</div>
`;
